test/e2e_flags.go

```diff
@@ -19,30 +19,94 @@ limitations under the License.
 package test
 
 import (
+	"context"
 	"flag"
+	"fmt"
+	"os"
+	"strings"
 
+	"github.com/knative/pkg/logging"
 	pkgTest "github.com/knative/pkg/test"
-	"github.com/knative/pkg/test/logging"
+	testLogging "github.com/knative/pkg/test/logging"
 )
 
-// EventingFlags holds the command line flags specific to knative/eventing
+const (
+	// E2ETestNamespacePrefix is the namespace prefix used for running all e2e tests.
+	E2ETestNamespacePrefix = "e2e-ns"
+	// DefaultClusterChannelProvisioner is the default ClusterChannelProvisioner we will run tests against.
+	DefaultClusterChannelProvisioner = "in-memory-channel"
+	// DefaultBrokerName is the name of the Broker that is automatically created after the current namespace is labeled.
+	DefaultBrokerName = "default"
+
+	appName = "eventing-e2e-testing"
+)
+
+// validProvisioners is a list of provisioners that Eventing currently support.
+var validProvisioners = []string{DefaultClusterChannelProvisioner}
+
+func isValid(provisioner string) bool {
+	for i := range validProvisioners {
+		if provisioner == validProvisioners[i] {
+			return true
+		}
+	}
+	return false
+}
+
+// EventingFlags holds the command line flags specific to knative/eventing.
 var EventingFlags = initializeEventingFlags()
 
-// EventingEnvironmentFlags holds the e2e flags needed only by the eventing repo
+// Provisioners holds the ClusterChannelProvisioners we want to run test against.
+type Provisioners []string
+
+func (ps *Provisioners) String() string {
+	return fmt.Sprint(*ps)
+}
+
+// Set converts the input string to Provisioners.
+// The default CCP we will test against is in-memory-channel.
+func (ps *Provisioners) Set(value string) error {
+	for _, provisioner := range strings.Split(value, ",") {
+		provisioner := strings.TrimSpace(provisioner)
+		if !isValid(provisioner) {
+			logger := logging.FromContext(context.Background()).Named(appName)
+			logger.Fatalf("The given provisioner %q is not supported, tests cannot be run.\n", provisioner)
```

Comment:
1. You don't need `os.Exit(1)` anymore.
1. Either make `logger` a global var, or create a helper for Fatalf.

Reply:
Done.
Made `logger` a global variable.